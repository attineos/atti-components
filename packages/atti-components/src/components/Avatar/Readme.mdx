---
path: '/avatar'
name: 'Avatar'
title: 'Avatar component'
---

Small Avatar :

```js
<Avatar size="small" bgColor="#4285F4">
  ML
</Avatar>
```

Medium Avatar :

```js
<Avatar bgColor="neutral">MV</Avatar>
```

Large Avatar :

```js
<Avatar size="large" bgColor="#FFC82C">
  EL
</Avatar>
```

Pictured Avatar :

```js
<Avatar
  bgSrc="https://image.flaticon.com/icons/svg/149/149071.svg"
  size="large"
  title="Eric Legrand"
/>
```
