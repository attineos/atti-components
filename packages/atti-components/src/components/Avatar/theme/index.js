export default ({ borders, colors, fonts, spaces }) => ({
  border: {
    radius: borders.radius.full,
  },
  colors: {
    text: colors.white,
  },
  fonts: {
    fontFamily: fonts.fontFamily.primary,
  },
  sizes: {
    small: {
      size: spaces.s1_5,
      fontSize: fonts.fontSize.fs12,
      lineHeight: fonts.lineHeight.lh10,
    },
    medium: {
      size: spaces.s2,
      fontSize: fonts.fontSize.fs16,
      lineHeight: fonts.lineHeight.lh16,
    },
    large: {
      size: spaces.s4,
      fontSize: fonts.fontSize.fs30,
      lineHeight: fonts.lineHeight.lh32,
    },
  },
})
