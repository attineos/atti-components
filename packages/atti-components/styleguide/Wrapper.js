import React from 'react'

import ThemeProvider from '../src/components/ThemeProvider'
import { defaultTheme } from '../src/themes'

export default ({ children }) => <ThemeProvider otherTheme={defaultTheme}>{children}</ThemeProvider>
