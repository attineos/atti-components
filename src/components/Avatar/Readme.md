Small Avatar :

```js
<Avatar name="Marc Lenormand Legrain" size="small" color="#4285F4" />
```

Medium Avatar :

```js
<Avatar name="Michel Varnier" />
```

Large Avatar :

```js
<Avatar name="Eric Legrand" size="large" color="#FFC82C" />
```

Pictured Avatar :

```js
<Avatar
  name="Eric Legrand"
  size="large"
  src="https://image.flaticon.com/icons/svg/149/149071.svg"
/>
```
