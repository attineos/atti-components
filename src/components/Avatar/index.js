import React from 'react'
import PropTypes from 'prop-types'
import { noop } from 'lodash'
import StyledAvatar from './styles/StyledAvatar'
import StyledPicturedAvatar from './styles/StyledPicturedAvatar'
import { abbrev_name } from '../../helpers/StringHelper/StringHelper'

/**
 * Avatar component
 *
 * @version 0.1
 * @author Adrien TINEL
 */

class Avatar extends React.Component {
  static defaultProps = {
    color: noop(),
    size: 'medium',
    src: noop(),
  }

  static propTypes = {
    /**
     * A deterministic color background
     */
    color: PropTypes.string,

    /**
     * Full name
     */
    name: PropTypes.string.isRequired,

    /**
     * Defines the size of the component
     */
    size: PropTypes.string,

    /**
     * Route to the background picture
     */
    src: PropTypes.string,
  }

  render() {
    const { name, size, src, color } = this.props

    // background of the avatar component
    let children
    if (src) {
      // pick the img route to define it as the background
      children = (
        <StyledPicturedAvatar
          alt={name}
          color={color}
          className="Avatar--img"
          size={size}
          src={src}
        />
      )
    } else {
      // initials as the background
      const abbr = abbrev_name(name)
      children = abbr
    }

    return (
      <StyledAvatar className="Avatar" size={size} name={name} color={color}>
        {children}
      </StyledAvatar>
    )
  }
}

export default Avatar
