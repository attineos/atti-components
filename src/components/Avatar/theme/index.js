export default ({ borders, colors, fonts, spaces }) => ({
  border: {
    radius: borders.radius.brCircle,
  },
  colors: {
    text: colors.white,
  },
  fonts: {
    fontFamily: fonts.fontFamily.primary,
  },
  sizes: {
    small: {
      size: spaces.s10,
      fontSize: fonts.fontSize.fs10,
      lineHeight: fonts.lineHeight.lh10,
    },
    medium: {
      size: spaces.s14,
      fontSize: fonts.fontSize.fs16,
      lineHeight: fonts.lineHeight.lh16,
    },
    large: {
      size: spaces.s18,
      fontSize: fonts.fontSize.fs32,
      lineHeight: fonts.lineHeight.lh32,
    },
  },
})
