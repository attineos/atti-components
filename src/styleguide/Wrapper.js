import React from 'react'
import PropTypes from 'prop-types'
import { remove, get } from 'lodash'

import Cookies from './helpers/Cookies'
import themes from './config/themes'

import ThemeProvider from '../components/ThemeProvider'

const updateWrappers = []

export const getWrappersToUpdate = () => updateWrappers

class Wrapper extends React.Component {
  static propTypes = {
    /**
     * The child node.
     */
    children: PropTypes.node.isRequired,
  }

  state = {
    selectedTheme: null,
  }

  componentWillMount() {
    updateWrappers.push(this)

    const selectedTheme = Cookies.hasItem('selectedTheme')
      ? Cookies.getItem('selectedTheme')
      : 'default'

    this.state.selectedTheme = get(themes, selectedTheme, null)
  }

  componentWillUnmount() {
    remove(updateWrappers, item => item === this)
  }

  handleSetTheme = newTheme =>
    this.setState(
      {
        selectedTheme: newTheme,
      },
      () => this.forceUpdate(),
    )

  render() {
    const { children } = this.props
    const { selectedTheme } = this.state

    return (
      <ThemeProvider theme={selectedTheme} key={Math.random()}>
        {children}
      </ThemeProvider>
    )
  }
}

export default Wrapper
