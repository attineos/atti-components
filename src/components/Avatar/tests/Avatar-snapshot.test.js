/* This file was generated automatically, please do not write in it. */
import React from 'react';
import Avatar from '..';
import Wrapper from 'styleguide/Wrapper'
import renderer from 'react-test-renderer';

const mock = () => {}

it('Avatar renders correctly', () => {
  const tree = renderer
    .create(
      <Wrapper>
        <Avatar color="Atticomponents" name="Atticomponents" size="Atticomponents" src="Atticomponents" ></Avatar>
      </Wrapper>
      )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
    