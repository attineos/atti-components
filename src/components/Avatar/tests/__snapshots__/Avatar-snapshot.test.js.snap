// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly 1`] = `
<div
  className="Avatar StyledAvatar-sc-8wohen-0 bDKzTQ"
  color="Atticomponents"
  name="Atticomponents"
  size="Atticomponents"
>
  <img
    alt="Atticomponents"
    className="Avatar--img StyledPicturedAvatar-sc-2gx68i-0 dVzuMq"
    color="Atticomponents"
    size="Atticomponents"
    src="Atticomponents"
  />
</div>
`;
